### PR TITLE
New Problem: 2D translation matrix

### PR DESCRIPTION
**Pull request by Mohamad-Abdulkadir**
Adding the implementation of the 2D Translation Matrix.

1. solution.py: includes the implementation of the problem with a detailed solution as well as two test cases.
2. learn.html: introduces the problem and the mathematical formula to solve it.

### CODE REVIEW

File: Problems/56_2D_translation_matrix/solution.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    
+    # Translation matrix
+    translation_matrix = np.array([
+        [1, 0, tx],
+        [0, 1, ty],
+        [0, 0, 1]
+    ])
+    
+    # Convert points to homogeneous coordinates
+    homogeneous_points = np.hstack([np.array(points), np.ones((len(points), 1))])
```
Comment (moe18) [inline Problems/56_2D_translation_matrix/solution.py:21]: id explain this in the learn section

File: Problems/56_2D_translation_matrix/learn.html [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+0 & 0 & 1
+\end{bmatrix}
+\]
+</p>
+
+<p>To apply this translation to a point (x, y) we use homogeneous coordinates by representing the point as (x, y, 1). Then the transformation is performed by matrix multiplication:</p>
+
+<p>
+\[
+\begin{bmatrix}
```
Comment (moe18) [inline Problems/56_2D_translation_matrix/learn.html:22]: id use `x` sub `t` this way people dont think it is `x` times `t`

**Review by moe18 — COMMENTED** [review]
looks good, had a few comments

**Comment by Mohamad-Abdulkadir** [comment]
The learn section is updated now, hopefully the comments above have been addressed.

**Review by moe18 — APPROVED** [review]

**Pull request merged**